File: JobApplicationTracker.Api/DTOs/JobApplication/CreateJobApplicationDto.cs
using JobApplicationTracker.Enums;

namespace JobApplicationTracker.DTOs;

public class CreateJobApplicationDto
{
    public string Company { get; set; }
    public string Position { get; set; }
    public string? Note { get; set; }
    public CreateJAStatusEntryDto InitialStatus { get; set; }
    public string? JobDescription { get; set; }
    public CreateJAEventDto? JAEvent { get; set; }
}

File: JobApplicationTracker.Frontend/types/JAObjects/JobApplications/CreateJobApplicationDto.ts
import type { CreateJAEventDto } from '../JAEvents/CreateJAEventDto';
import type { CreateInitJAStatusDto } from '../JAStatuses/CreateInitJAStatusDto';

export interface CreateJobApplicationDto {
  company: string;
  position: string;
  note: string | null;
  initialStatus: CreateInitJAStatusDto;
  jobDescription: string | null;
  jaEvent: CreateJAEventDto | null;
}